decisions/0004-remixing-react-router.md

```diff
@@ -0,0 +1,173 @@
+
+### `<Form method="get">` is no longer a "submission"
+
+Functionally, these two bits of code are identical, with the only difference being that in the `<form>` case you let the user determine the query value.
+
+```jsx
+<a href="/search?query=matt">Search</a>
+
+<form action="/search">
+  <input name="query" value="matt" />
+  <button type="submit">Search</button>
+</form>
+```
+
+But, in Remix we were considering the latter a "submission" such that `useTransition().state === "submitting"`. In order to ensure our "navigations" reflect the browser behavior we have changed this in the router such that GET Form submissions result in `useNavigation().state === "loading"`.
+
+**Backwards Compatibility**
+
+This will be handled in the deprecated `useTransition` hook along with the backfill of `type` and `submission` properties
+
+### `unstable_shouldReload` stabilized as `shouldRevalidate`
+
+We stabilized the API for when a given route loader should re-run, and changed the name to align with the "revalidation" nomenclature and the `useRevalidator` hook. We also leave more control in the hands of the user here. In Remix there were some cases in which you _could not_ opt out of revalidation and if your method did run you had full control and couldn't necessarily handle one edge case and then say "do what you otherwise would have done".
+
+Now, if you provide a `shouldRevalidate` method we will call it during all revalidations and provide you a `defaultShouldRevalidate` boolean value. This allows you to opt out of any revalidation, and also code your own logic to fallback on our default choice:
+
+```js
+function shouldRevalidate({ defaultShouldRevalidate }) {
+  // Don't revalidate for this case
+  if (someEdgeCase()) {
+    return false;
+  }
+
+  // Otherwise, do what we would have done by default
+  return defaultShouldRevalidate;
+}
+```
+
+### `<ScrollRestoration getKey>` prop
+
+In Remix, the `<ScrollRestoration>` component made an assumption that we would always restore scroll position based on `location.key`. If the key was the same as a prior location we knew the scroll position for, then we knew you had been there before and we should restore. This works great for back/forward navigations but it's a bit overly restrictive. You cannot choose tgo restore scroll based on anything other than `key`. Twitter has a great implementation of this as you click around in their left nav bar - your tweet feed is always at the same place when you click back to it - even though it's a _new_ location in the history stack. This is because they're restoring by pathname here instead of `location.key`. Or maybe you want to maintain scroll position for all routes under a given pathname and you thus want to use a portion of the pathname as the scroll restoration key.
```

Comment:
```suggestion
In Remix, the `<ScrollRestoration>` component made an assumption that we would always restore scroll position based on `location.key`. If the key was the same as a prior location we knew the scroll position for, then we knew you had been there before and we should restore. This works great for back/forward navigations but it's a bit overly restrictive. You cannot choose to restore scroll based on anything other than `key`. Twitter has a great implementation of this as you click around in their left nav bar - your tweet feed is always at the same place when you click back to it - even though it's a _new_ location in the history stack. This is because they're restoring by pathname here instead of `location.key`. Or maybe you want to maintain scroll position for all routes under a given pathname and you thus want to use a portion of the pathname as the scroll restoration key.
```